### ISSUE TITLE
plotly heatmap doesn't take a dataframe

### ISSUE THREAD

**Issue by ziyadsaeed** [ropensci/plotly#329]
The figure reference documentation says 
```r
plot_ly(df, type="heatmap"[, ...])
```
the df in the code above indicates a data frame 
The code below doesn't work

```r
url <- "http://datasets.flowingdata.com/ppg2008.csv"
nba_players <- read.csv(url, row.names = 1)

plot_ly(nba_players, type = "heatmap")
```

A solution is to use a data.matrix
```r
plot_ly(z = data.matrix(nba_players), 
           x = colnames(nba_players), 
           y = row.names(nba_players), 
           type = "heatmap")
```

**Comment by ziyadsaeed** [comment]
what then is the syntax to use a dataframe without converting it to a matrix

**Comment by cpsievert** [comment]
I don't think it is currently possible to make a heatmap from a data frame. This will go on the todo list, but it isn't a super high priority right now.

PS. you can adds axis titles like so:

```r
plot_ly(z = volcano, type = "heatmap") %>% layout(xaxis = list(title = "x"), yaxis = list(title = "y"))
```

**Comment by ziyadsaeed** [comment]
for the time being this part of the heatmap documentation should be changed. Atleast it should mention that dataframe support is in the works

 y (dataframe column, list, vector)
Sets the y coordinates.
x (dataframe column, list, vector)
Sets the x coordinates.
z (dataframe column, list, vector)
Sets the z data.

**Comment by simonegabbriellini** [comment]
@ziyadsaeed are you sure about your data.matrix solution? if I run your code, I get:

     plot_ly(z = data.matrix(nba_players), 
    +         x = colnames(nba_players), 
    +         y = row.names(nba_players), 
    +         type = "heatmap")
    Error: Variables must be length 1 or 50. 
    Problem variables: 'x'

Looks like the axes should have same length, otherwise it complains... I don't understand why though...
